externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala

```diff
@@ -142,7 +151,18 @@ class ExecuteStatement(
           case TypedResult.ResultType.PAYLOAD =>
             (1 to result.getPayload).foreach { page =>
               if (rows.size < resultMaxRows) {
-                rows ++= executor.retrieveResultPage(resultId, page).asScala
+                val result = executor.retrieveResultPage(resultId, page).asScala.toList
```

Comment:
Would you please add comments here? e.g. FLINK-xxx changes the return type from xx to yy